src/GraphQL/Types/InputObjectGraphType.cs

```diff
@@ -14,13 +14,40 @@ public class InputObjectGraphType<TSourceType> : ComplexGraphType<TSourceType>,
     {
         public override FieldType AddField(FieldType fieldType)
         {
-            if (fieldType.Type == typeof(ObjectGraphType))
+            if (!ValidateFieldType(fieldType))
             {
                 throw new ArgumentException("InputObjectGraphType cannot have fields containing a ObjectGraphType.", nameof(fieldType.Type));
```

Comment:
```suggestion
                 throw new ArgumentException("InputObjectGraphType should contain only fields of 'input types' - enumerations, scalars and other InputObjectGraphTypes", nameof(fieldType.Type));
```

This error message is more informative.